app/controllers/complaints_controller.rb

```diff
@@ -1,5 +1,20 @@
 class ComplaintsController < ApplicationController
+  before_action :check_pa11y_id, only: :show, if: -> { Rails.env.test? }
+
   def index
     @complaints = FakeData::ApiResponse.hses_issues_response[:data]
   end
+
+  def show
+    @complaint = FakeData::ApiResponse.hses_issue_response(params[:id])[:data]
+    render layout: "details"
+  end
+
+  private
+
+  def check_pa11y_id
+    if params[:id] == "pa11y-id"
+      params[:id] = FakeData::ApiResponse.hses_issues_response[:data].first[:id]
```

Comment:
Added this special case since we won't know what IDs we have available on a consistent basis. If this is too messy, I suppose we could also ensure that the stack of 25 issues always has one particular ID present, like "1" always being in the list.